packages/core/http/core-http-router-server-internal/src/request.ts

```diff
@@ -120,9 +120,9 @@ export class CoreKibanaRequest<
 
   constructor(
     request: RawRequest,
-    public readonly params: Params,
-    public readonly query: Query,
-    public readonly body: Body,
+    public params: Params,
+    public query: Query,
+    public body: Body,
```

Comment:
Not super happy about this, but it seemed like the least trouble without doing one of:

(1) Exposing more knowledge of `Router` internals so that we can dynamically set validation schema (right now we just use `IRouter`)
(2) Changing our approach, integrating the versioned router functionality into the `Router` implementation more directly

Reply:
Also, the `KibanaRequest` type still presents these values as `readonly`.